Dockerfile

```diff
@@ -2,23 +2,38 @@
 
 ARG base_image="debian:bullseye-slim"
 
+FROM scratch as frontend-source
+
+# artisan is not strictly required for JS builds but it triggers some behavior inside Laravel Mix
+# https://github.com/laravel-mix/laravel-mix/issues/1326#issuecomment-363975710
+COPY --link package.json package-lock.json webpack.mix.js artisan /app/
```

Comment:
purely out of curiosity since I can't seem to find it in the docs, what does the `--link` here do?

Reply:
It is hard to find (it's [here](https://hub.docker.com/r/docker/dockerfile#:~:text=Linked%20copies%20COPY%20%2D%2Dlink%2C%20ADD%20%2D%2Dlink)), but basically it improves caching across builds.